### PR TITLE
Handles '/' and '\' in PhysicalFileWatcher.IsDirectoryPath

### PR DESCRIPTION
**Pull request by hishamco**
@glennc based on your issue #47

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@hishamco__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSOTBOT;

**Comment by glennc** [comment]
Thanks for the PR @hishamco. It looks OK to me. @muratg, @davidfowl, @Praburaj can one of you look at this as well? I don't remember exactly how we are using this, so someone else should give the shipit :)

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs [hunk 1]
```diff
@@ -148,10 +148,10 @@ private string NormalizeFilter(string filter)
...
         private bool IsDirectoryPath(string path)
         {
-            return path != null && path.Length > 1 &&
-                (path[path.Length - 1] == Path.DirectorySeparatorChar ||
-                path[path.Length - 1] == Path.AltDirectorySeparatorChar);
-        }
+			if (path == null)
+				throw new ArgumentNullException();
+			return path.EndsWith(Path.DirectorySeparatorChar.ToString()) || path.EndsWith(Path.AltDirectorySeparatorChar.ToString());
+		}
```
Comment (troydai) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: Nit: place use space to replace all tabs.

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs [hunk 2]
```diff
@@ -148,10 +148,10 @@ private string NormalizeFilter(string filter)
 
         private bool IsDirectoryPath(string path)
         {
-            return path != null && path.Length > 1 &&
-                (path[path.Length - 1] == Path.DirectorySeparatorChar ||
-                path[path.Length - 1] == Path.AltDirectorySeparatorChar);
-        }
+			if (path == null)
+				throw new ArgumentNullException();
```
Comment (troydai) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: Well, this is a behavior change. `ArgumentNullException` wasn't thrown previously.

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs [hunk 3]
```diff
@@ -148,10 +148,10 @@ private string NormalizeFilter(string filter)
...
+			return path.EndsWith(Path.DirectorySeparatorChar.ToString()) || path.EndsWith(Path.AltDirectorySeparatorChar.ToString());
```
Comment (troydai) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: Can we avoid the two `ToString()` to save the allocation?

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs (hunk 2 above)
Comment (hishamco) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: @troydai Can we use the NotNull attribute instead?

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs (hunk 1 above)
Comment (hishamco) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: @troydai thanks I will do that soon

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs (hunk 2 above)
Comment (troydai) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: Hi @hishamco. The point is not about how the `ArgumentNullException` be thrown but about whether it should be thrown. You see previously the code return `false` when the `path` is null. The client codes may rely on this behavior. Now if the contract changed to thrown exception, then it's a behavior change and it's a breaking change.

Comment (hishamco) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: @troydai thanks, I will remove the ArgumentNullException and replace it with null check

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs (hunk 3 above)
Comment (hishamco) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: @troydai I think adding '=' in path.Length>=1 in the old code will be the suitable solution

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs (hunk 2 above)
Comment (troydai) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: Thank you :)

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs (hunk 3 above)
Comment (troydai) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: Yup. I think so.

**Comment by hishamco** [comment]
@troydai Done!!

**Comment by troydai** [comment]
Thanks @hishamco. It looks good to me. 

I'd like bring another pairs of eyes to take another look. 
@glennc @Praburaj

**Comment by muratg** [comment]
Looks good to me as well.  :shipit:

**Comment by muratg** [comment]
@troydai can you rebase and take this one?

**Comment by troydai** [comment]
On it.

**Comment by troydai** [comment]
@hishamco your change has been merged. Thanks for the contribution!

**Comment by hishamco** [comment]
You are welcome

**Comment by Eilon** [comment]
@hishamco @troydai so can https://github.com/aspnet/FileSystem/issues/47 be closed?

**Comment by hishamco** [comment]
Of course @Eilon , thanks for your reminder :)

**Pull request closed without merging**